nixos/modules/services/web-apps/rimgo.nix

```
@@ -0,0 +1,94 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}: let
+  cfg = config.services.rimgo;
+  inherit (lib)
+    mkOption
+    mkEnableOption
+    mkPackageOption
+    mkDefault
+    mkIf
+    types
+    literalExpression
+  ;
+in {
+  options.services.rimgo = {
+    enable = mkEnableOption "rimgo";
+    package = mkPackageOption pkgs "rimgo" {};
+    settings = mkOption {
+      type = with types; attrsOf str;
+      default = {};
+      example = literalExpression ''
+        {
+          FORCE_WEBP = "1";
+        }
+      '';
+      description = ''
+        Settings for rimgo, see [the official documentation](https://rimgo.codeberg.page/docs/usage/configuration/) for supported options.
+      '';
+    };
+    port = mkOption {
+      type = types.port;
+      default = 3000;
+      example = 69420;
+      description = "The port to use.";
+    };
+    address = mkOption {
+      type = types.str;
+      default = "127.0.0.1";
+      description = "The address to listen on.";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.rimgo.settings = {
+      PORT = mkDefault (toString cfg.port);
```

```suggestion
      PORT = mkDefault cfg.port;
```
We can do this if using the coerced type for `settings`.